url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/visualisation/charts/Generic_AgeGenderLineChart.java

```diff
@@ -0,0 +1,870 @@
+            min = firstOrderStats[6];
+                    q1 = firstOrderStats[3];
+            median = firstOrderStats[2];
+            q3 = firstOrderStats[4];
+            max = firstOrderStats[7];
+            int minPointCol = coordinateToScreenCol(min.negate());
+            int q1PointCol = coordinateToScreenCol(q1.negate());
+            int medianPointCol = coordinateToScreenCol(median.negate());
+            int q3PointCol = coordinateToScreenCol(q3.negate());
+            int maxPointCol = coordinateToScreenCol(max.negate());
+            int pointRow = coordinateToScreenRow(
+                    BigDecimal.valueOf(age - ageInterval / 2));
+            if (firstPoint) {
+                last_minPointCol = minPointCol;
+                last_q1PointCol = q1PointCol;
+                last_medianPointCol = medianPointCol;
+                last_q3PointCol = q3PointCol;
+                last_maxPointCol = maxPointCol;
+                lastPointRow = pointRow;
+                firstPoint = false;
+            }
+            // Draw min line
+            abLine2D = new Line2D.Double(
+                    last_minPointCol,
+                    lastPointRow,
+                    minPointCol,
+                    pointRow);
+            setPaint(Color.LIGHT_GRAY);
+            draw(abLine2D);
+            // Draw max line
+            abLine2D = new Line2D.Double(
+                    last_maxPointCol,
+                    lastPointRow,
+                    maxPointCol,
+                    pointRow);
+            setPaint(Color.LIGHT_GRAY);
+            draw(abLine2D);
+            // Draw q1 line
+            abLine2D = new Line2D.Double(
+                    last_q1PointCol,
+                    lastPointRow,
+                    q1PointCol,
+                    pointRow);
+            setPaint(Color.GRAY);
+            draw(abLine2D);
+            // Draw q3 line
+            abLine2D = new Line2D.Double(
+                    last_q3PointCol,
+                    lastPointRow,
+                    q3PointCol,
+                    pointRow);
+            setPaint(Color.GRAY);
+            draw(abLine2D);
+            // Draw median line
+            abLine2D = new Line2D.Double(
+                    last_medianPointCol,
+                    lastPointRow,
+                    medianPointCol,
+                    pointRow);
+            setPaint(Color.DARK_GRAY);
+            draw(abLine2D);
+            // Set parameters
+            last_minPointCol = minPointCol;
+            last_q1PointCol = q1PointCol;
+            last_medianPointCol = medianPointCol;
+            last_q3PointCol = q3PointCol;
+            last_maxPointCol = maxPointCol;
+            lastPointRow = pointRow;
+        }
+    }
+
+    /**
+     * Override this method to use other data
+     *
+     * @param femalePopAge0
+     * @param malePopAge0
+     * @return
+     */
+    private static Object[] getPopulationData(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 513-555, lines 556-598 into separate methods.